ff_stdio.c

```diff
@@ -88,7 +88,7 @@ int prvFFErrorToErrno( FF_Error_t xError );
     typedef struct WORKING_DIR
     {
         char pcCWD[ ffconfigMAX_FILENAME ];      /* The current working directory. */
-        char pcFileName[ ffconfigMAX_FILENAME ]; /* The created absolute path. */
+        char pcFileName[ ffconfigMAX_FILENAME+1 ]; /* The created absolute path. */
```

Comment:
@aggarg : +1

One comment about the use of `ffconfigMAX_FILENAME`: since the beginning this macro defines the maximum length of a filename including the terminating zero.

Please also have a look at this [PR #28](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-FAT/pull/28)

In that PR it was decided to keep the same interpretation of `ffconfigMAX_FILENAME`, see for instance this change:
~~~diff
-STRNCPY( pcFileName, ( pcPath + xIndex + 1 ), ffconfigMAX_FILENAME );
+STRNCPY( pcFileName, ( pcPath + xIndex + 1 ), ffconfigMAX_FILENAME - 1 );
~~~
And finally this comment:
~~~c
/* Sets the maximum length for file names, including the path.
 * Note that the value of this define is directly related to the maximum stack
 * use of the +FAT library. In some API's, a character buffer of size
 * 'ffconfigMAX_FILENAME' will be declared on stack. As such, this value should
 * include space for the null byte. */
#define ffconfigMAX_FILENAME    129
~~~